compiler/damlc/tests/BUILD.bazel

```diff
@@ -1000,3 +1000,21 @@ daml_compile_with_dalf(
     # PlatformIndependence.daml.
     ghc_options = [],
 )
+
+# Test for interfaces + data-dependencies
+daml_build_test(
+    name = "interface-definition",
+    project_dir = "interfaces-data-deps/interface-definition",
+)
+
+daml_test(
```

Comment:
To start with, I'm not even attempting to import the project above as a data-dep, just as a plain dep, but `bazel test //compiler/damlc/tests:interface-implementation` fails with 

```
  compiler/damlc/tests/interfaces-data-deps/interface-implementation/daml/Asset.daml:3:1: error:
  Could not find module ‘Token’
  There are files missing in the ‘interface-definition-0.0.0’ package,
  try running 'ghc-pkg check'.
  Use -v (or `:set -v` in ghci) to see a list of the files searched for.
```

so I'm probably doing something wrong above. 

I'm basing this on a test in `docs/` (see [docs/BUILD.bazel#L646-L667](https://github.com/digital-asset/daml/blob/main/docs/BUILD.bazel#L646-L667)), where a `daml_test` has a `data_deps` that corresponds to the `.dar` generated by a `daml_build_test`, but I probably messed something up along the way